suite/tests/client-interface/count-ctis.c

```diff
@@ -95,5 +96,35 @@ GLOBAL_LABEL(FUNCNAME:)
 
 
 END_FILE
+#elif defined(AARCH64)
+START_FILE
+
+#define FUNCNAME test_jecxz
+        DECLARE_FUNC(FUNCNAME)
+GLOBAL_LABEL(FUNCNAME:)
+        mov      x1, ARG1
+        END_PROLOG
+
+        /* test cbnz */
+        mov      x1, 0
+        cbz      x1, jecxz_taken
+        nop
+     jecxz_taken:
+        mov      x1, 1
+        cbz      x1, jecxz_taken
+        nop
+     jecxz_not_taken:
```

Comment:
Please add a test case here for each condition in your code:

+ cbz using stolen register
+ cbnz
+ cbnz using stolen register
+ tbz
+ tbz using stolen register
+ tbnz
+ bcond

(Even without a comprehensive register state check before vs after each, which is not simple to add, just having these exercise each part of the instrumentation code is a good step.  And errors in this type of instrumentation often result in visible crashes.  Could add some aarch64-specific testing in the client for further tests but that is optional.  Just adding app code for each of the cases listed should be simple and go a long way.)

Reply:
Done.